--- a/configure.ac	2015-12-05 19:10:40.000000000 +0300
+++ a/configure.ac	2015-12-05 19:09:57.000000000 +0300
@@ -8,4 +8,12 @@
 AC_PROG_MAKE_SET
 
+AC_ARG_ENABLE(oldname-compat,
+ AS_HELP_STRING([--disable-oldname-compat],
+   [Don't include the old libjson.so library and include/json directory.]),
+[],
+[enable_oldname_compat=yes]
+)
+AM_CONDITIONAL(ENABLE_OLDNAME_COMPAT, [test "x${enable_oldname_compat}" != "xno"])
+
 AC_ARG_ENABLE(rdrand,
  AS_HELP_STRING([--enable-rdrand],
@@ -93,4 +101,5 @@
 AC_CONFIG_FILES([
 Makefile
+json.pc
 json-c.pc
 tests/Makefile
