--- a/configure	2014-04-11 05:48:23.000000000 +0400
+++ b/configure	2015-12-05 18:39:26.126288468 +0300
@@ -687,6 +687,8 @@
 AM_DEFAULT_VERBOSITY
 AM_DEFAULT_V
 AM_V
+ENABLE_OLDNAME_COMPAT_FALSE
+ENABLE_OLDNAME_COMPAT_TRUE
 am__untar
 am__tar
 AMTAR
@@ -752,6 +754,7 @@
 ac_user_opts='
 enable_option_checking
 enable_silent_rules
+enable_oldname_compat
 enable_rdrand
 enable_dependency_tracking
 enable_shared
@@ -1391,6 +1394,9 @@
   --disable-FEATURE       do not include FEATURE (same as --enable-FEATURE=no)
   --enable-FEATURE[=ARG]  include FEATURE [ARG=yes]
   --enable-silent-rules   less verbose build output (undo: "make V=1")
+  --disable-oldname-compat
+                          Don't include the old libjson.so library and
+                          include/json directory.
   --disable-silent-rules  verbose build output (undo: "make V=0")
   --enable-rdrand         Enable RDRAND Hardware RNG Hash Seed generation on
                           supported x86/x64 platforms.
@@ -2948,6 +2954,7 @@
 fi
 
 
+
 # Check whether --enable-rdrand was given.
 if test "${enable_rdrand+set}" = set; then :
   enableval=$enable_rdrand; if test x$enableval = xyes; then
@@ -2958,6 +2965,21 @@
 fi
 fi
 
+# Check whether --enable-oldname-compat was given.
+if test "${enable_oldname_compat+set}" = set; then :
+  enableval=$enable_oldname_compat;
+else
+  enable_oldname_compat=yes
+
+fi
+
+ if test "x${enable_oldname_compat}" != "xno"; then
+  ENABLE_OLDNAME_COMPAT_TRUE=
+  ENABLE_OLDNAME_COMPAT_FALSE='#'
+else
+  ENABLE_OLDNAME_COMPAT_TRUE='#'
+  ENABLE_OLDNAME_COMPAT_FALSE=
+fi
 
 if test "x$enable_rdrand" = "xyes"; then
   { $as_echo "$as_me:${as_lineno-$LINENO}: result: RDRAND Hardware RNG Hash Seed enabled on supported x86/x64 platforms" >&5
@@ -12233,6 +12255,10 @@
   am__EXEEXT_FALSE=
 fi
 
+if test -z "${ENABLE_OLDNAME_COMPAT_TRUE}" && test -z "${ENABLE_OLDNAME_COMPAT_FALSE}"; then
+  as_fn_error $? "conditional \"ENABLE_OLDNAME_COMPAT\" was never defined.
+Usually this means the macro was only invoked conditionally." "$LINENO" 5
+fi
 if test -z "${AMDEP_TRUE}" && test -z "${AMDEP_FALSE}"; then
   as_fn_error $? "conditional \"AMDEP\" was never defined.
 Usually this means the macro was only invoked conditionally." "$LINENO" 5
