--- a/Makefile.in	2015-12-05 18:05:23.000000000 +0300
+++ b/Makefile.in	2015-12-05 18:20:33.943172112 +0300
@@ -88,5 +88,8 @@
 	depcomp $(libjson_cinclude_HEADERS) COPYING compile \
 	config.guess config.sub install-sh missing ltmain.sh
+@ENABLE_OLDNAME_COMPAT_TRUE@am__append_1 = libjson.la
+@ENABLE_OLDNAME_COMPAT_TRUE@am__append_2 = json.pc
+
 subdir = .
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
 am__aclocal_m4_deps = $(top_srcdir)/configure.ac
@@ -97,7 +100,7 @@
  configure.lineno config.status.lineno
 mkinstalldirs = $(install_sh) -d
 CONFIG_HEADER = config.h json_config.h
-CONFIG_CLEAN_FILES = json-c.pc json-c-uninstalled.pc
+CONFIG_CLEAN_FILES = json.pc json-c.pc json-c-uninstalled.pc
 CONFIG_CLEAN_VPATH_FILES =
 am__vpath_adj_setup = srcdirstrip=`echo "$(srcdir)" | sed 's|.|.|g'`;
 am__vpath_adj = case $$p in \
@@ -141,6 +144,13 @@
 libjson_c_la_LINK = $(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) \
 	$(LIBTOOLFLAGS) --mode=link $(CCLD) $(AM_CFLAGS) $(CFLAGS) \
 	$(libjson_c_la_LDFLAGS) $(LDFLAGS) -o $@
+libjson_la_DEPENDENCIES =
+libjson_la_SOURCES = libjson.c
+libjson_la_OBJECTS = libjson.lo
+libjson_la_LINK = $(LIBTOOL) --tag=CC $(AM_LIBTOOLFLAGS) \
+	$(LIBTOOLFLAGS) --mode=link $(CCLD) $(AM_CFLAGS) $(CFLAGS) \
+	$(libjson_la_LDFLAGS) $(LDFLAGS) -o $@
+@ENABLE_OLDNAME_COMPAT_TRUE@am_libjson_la_rpath = -rpath $(libdir)
 AM_V_P = $(am__v_P_@AM_V@)
 am__v_P_ = $(am__v_P_@AM_DEFAULT_V@)
 am__v_P_0 = false
@@ -175,8 +185,8 @@
 am__v_CCLD_ = $(am__v_CCLD_@AM_DEFAULT_V@)
 am__v_CCLD_0 = @echo "  CCLD    " $@;
 am__v_CCLD_1 = 
-SOURCES = $(libjson_c_la_SOURCES)
-DIST_SOURCES = $(libjson_c_la_SOURCES)
+SOURCES = $(libjson_c_la_SOURCES) libjson.c
+DIST_SOURCES = $(libjson_c_la_SOURCES) libjson.c
 RECURSIVE_TARGETS = all-recursive check-recursive cscopelist-recursive \
 	ctags-recursive dvi-recursive html-recursive info-recursive \
 	install-data-recursive install-dvi-recursive \
@@ -409,6 +419,11 @@
 #libjsonx_include_HEADERS = \
 #	json_config.h
 libjson_c_la_LDFLAGS = -version-info 2:1:0 -no-undefined @JSON_BSYMBOLIC_LDFLAGS@
+@ENABLE_OLDNAME_COMPAT_TRUE@libjson_la_LDFLAGS = -version-info 1:0:1 -no-undefined -ljson-c
+
+# Temporary libjson library.  This will be removed after one release.
+@ENABLE_OLDNAME_COMPAT_TRUE@libjson_la_LIBADD = -ljson-c
+
 libjson_c_la_SOURCES = \
 	arraylist.c \
 	debug.c \
@@ -485,6 +500,8 @@
 
 distclean-hdr:
 	-rm -f config.h stamp-h1 json_config.h stamp-h2
+json.pc: $(top_builddir)/config.status $(srcdir)/json.pc.in
+	cd $(top_builddir) && $(SHELL) ./config.status $@
 json-c.pc: $(top_builddir)/config.status $(srcdir)/json-c.pc.in
 	cd $(top_builddir) && $(SHELL) ./config.status $@
 json-c-uninstalled.pc: $(top_builddir)/config.status $(srcdir)/json-c-uninstalled.pc.in
@@ -527,6 +544,9 @@
 
 libjson-c.la: $(libjson_c_la_OBJECTS) $(libjson_c_la_DEPENDENCIES) $(EXTRA_libjson_c_la_DEPENDENCIES) 
 	$(AM_V_CCLD)$(libjson_c_la_LINK) -rpath $(libdir) $(libjson_c_la_OBJECTS) $(libjson_c_la_LIBADD) $(LIBS)
+libjson.la: $(libjson_la_OBJECTS) $(libjson_la_DEPENDENCIES) $(EXTRA_libjson_la_DEPENDENCIES) 
+	$(AM_V_CCLD)$(libjson_la_LINK) $(am_libjson_la_rpath) $(libjson_la_OBJECTS) $(libjson_la_LIBADD) $(LIBS)
+
 
 mostlyclean-compile:
 	-rm -f *.$(OBJEXT)
@@ -541,6 +561,7 @@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/json_object_iterator.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/json_tokener.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/json_util.Plo@am__quote@
+@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libjson.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/linkhash.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/printbuf.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/random_seed.Plo@am__quote@
@@ -950,6 +971,8 @@
 maintainer-clean-generic:
 	@echo "This command is intended for maintainers to use"
 	@echo "it deletes files that may require special tools to rebuild."
+@ENABLE_OLDNAME_COMPAT_FALSE@uninstall-local:
+@ENABLE_OLDNAME_COMPAT_FALSE@install-data-hook:
 clean: clean-recursive
 
 clean-am: clean-generic clean-libLTLIBRARIES clean-libtool \
@@ -975,7 +998,8 @@
 info-am:
 
 install-data-am: install-libjson_cincludeHEADERS install-pkgconfigDATA
-
+	@$(NORMAL_INSTALL)
+	$(MAKE) $(AM_MAKEFLAGS) install-data-hook
 install-dvi: install-dvi-recursive
 
 install-dvi-am:
@@ -1060,9 +1084,13 @@
 maintainer-clean-local:
 	-rm -rf configure
 
-uninstall-local:
-	rm -rf "$(DESTDIR)@includedir@/json-c"
-	rm -f "$(DESTDIR)@includedir@/json"
+@ENABLE_OLDNAME_COMPAT_TRUE@install-data-hook:
+@ENABLE_OLDNAME_COMPAT_TRUE@	test \! -e "$(DESTDIR)@includedir@/json" || rm "$(DESTDIR)@includedir@/json"
+@ENABLE_OLDNAME_COMPAT_TRUE@	$(LN_S) json-c "$(DESTDIR)@includedir@/json"
+
+@ENABLE_OLDNAME_COMPAT_TRUE@uninstall-local:
+@ENABLE_OLDNAME_COMPAT_TRUE@	rm -f "$(DESTDIR)@includedir@/json"
+@ENABLE_OLDNAME_COMPAT_TRUE@	rm -rf "$(DESTDIR)@includedir@/json-c"
 
 Android.mk: Makefile.am
 	androgenizer -:PROJECT json-c \
