--- a/CMakeLists.txt	2014-11-15 16:49:31.000000000 +0800
+++ b/CMakeLists.txt	2015-11-27 21:08:25.966960023 +0800
@@ -5,7 +5,7 @@ PROJECT(uhttpd C)
 INCLUDE (CheckFunctionExists)
 
 SET(CMAKE_SHARED_LIBRARY_LINK_C_FLAGS "")
-ADD_DEFINITIONS(-D_FILE_OFFSET_BITS=64 -Os -Wall -Werror -Wmissing-declarations --std=gnu99 -g3)
+ADD_DEFINITIONS(-D_FILE_OFFSET_BITS=64 -Wall -Werror -Wmissing-declarations --std=gnu99)
 
 OPTION(TLS_SUPPORT "TLS support" ON)
 OPTION(LUA_SUPPORT "Lua support" ON)
