--- fstools-2015-11-22/libblkid-tiny/superblocks.h
+++ fstools-2015-11-22/libblkid-tiny/superblocks.h
@@ -94,7 +94,8 @@
 
 extern int blkid_probe_set_uuid(blkid_probe pr, unsigned char *uuid);
 extern int blkid_probe_set_uuid_as(blkid_probe pr, unsigned char *uuid, const char *name);
-
+extern size_t blkid_encode_to_utf8(int enc, unsigned char *dest, size_t len,
+			const unsigned char *src, size_t count);
 extern int blkid_probe_set_id_label(blkid_probe pr, const char *name,
 			     unsigned char *data, size_t len);
 extern int blkid_probe_set_utf8_id_label(blkid_probe pr, const char *name,

