--- a/Makefile	2016-01-22 14:06:42.000000000 +0300
+++ b/Makefile	2016-01-22 14:56:08.500895016 +0300
@@ -2,15 +2,13 @@
 CC 	=	gcc
 INSTALL	=	install
 IFLAGS  = -idirafter dummyinc
-#CFLAGS = -g
-CFLAGS	=	-O2 -fPIE -fstack-protector --param=ssp-buffer-size=4 \
-	-Wall -W -Wshadow -Werror -Wformat-security \
-	-D_FORTIFY_SOURCE=2 \
-	#-pedantic -Wconversion
 
-LIBS	=	-lcrypt -lnsl
-LINK	=
-LDFLAGS	=	-fPIE -pie -Wl,-z,relro -Wl,-z,now
+CFLAGS	+= -O2 -Wall -W -Werror -Wshadow -Wformat-security -D_FORTIFY_SOURCE=2
+CFLAGS	+= -ffunction-sections -fdata-sections
+
+LIBS	=	#-lcrypt -lnsl
+LINK	= -Wl,-s
+LDFLAGS	+= -Wl,--gc-sections
 
 OBJS	=	main.o utility.o prelogin.o ftpcmdio.o postlogin.o privsock.o \
 		tunables.o ftpdataio.o secbuf.o ls.o \
