--- a/libblkid-tiny/hfs.c	2015-11-27 14:24:54.000000000 +0300
+++ a/libblkid-tiny/hfs.c	2015-12-27 00:50:33.407871573 +0300
@@ -12,7 +12,7 @@
 #include <inttypes.h>
 
 #include "superblocks.h"
-//#include "md5.h"
+#include "md5.h"
 
 /* HFS / HFS+ */
 struct hfs_finder_info {
@@ -130,7 +130,7 @@
 
 static int hfs_set_uuid(blkid_probe pr, unsigned char const *hfs_info, size_t len)
 {
-/*	static unsigned char const hash_init[MD5LENGTH] = {
+	static unsigned char const hash_init[MD5LENGTH] = {
 		0xb3, 0xe2, 0x0f, 0x39, 0xf2, 0x92, 0x11, 0xd6,
 		0x97, 0xa4, 0x00, 0x30, 0x65, 0x43, 0xec, 0xac
 	};
@@ -145,7 +145,7 @@
 	MD5Final(uuid, &md5c);
 	uuid[6] = 0x30 | (uuid[6] & 0x0f);
 	uuid[8] = 0x80 | (uuid[8] & 0x3f);
-	return blkid_probe_set_uuid(pr, uuid);*/
+	return blkid_probe_set_uuid(pr, uuid);
 	return -1;
 }
 
