--- a/programs/_confread/Makefile
+++ b/programs/_confread/Makefile
@@ -14,7 +14,6 @@
 OPENSWANSRCDIR?=$(shell cd ../..; pwd)
 include ${OPENSWANSRCDIR}/Makefile.inc
 
-EXTRA5MAN=ipsec.conf.5
 CONFFILES=ipsec.conf
 EXCONFILES=ipsec.win2k.conf
 SOURCEDIR=_confread
@@ -24,11 +23,3 @@ CONFDFILES=clear clear-or-private privat
 
 include ${srcdir}../Makefile.program
 
-ipsec.conf.5: ${srcdir}ipsec.conf.5.xml
-
-${srcdir}ipsec.conf.5.xml: $(wildcard d.ipsec.conf/*.xml) ${srcdir}d.ipsec.conf/order.txt
-	cd ${srcdir} && cat `cat ${srcdir}d.ipsec.conf/order.txt` >ipsec.conf.5.xml
-
-diff:	ipsec.conf.5.xml
-	diff -B -w -u ipsec.conf.5.xml ipsec.conf.5.xml-ref
-
