--- ./drivers/net/wireless/ath/ath9k/init.c	2015-09-27 14:51:47.000000000 +0800
+++ ./drivers/net/wireless/ath/ath9k/init.c	2015-09-27 14:52:05.707875696 +0800
@@ -745,6 +745,7 @@ static void ath9k_set_hw_capab(struct at
 	hw->max_rate_tries = 10;
 	hw->sta_data_size = sizeof(struct ath_node);
 	hw->vif_data_size = sizeof(struct ath_vif);
+	hw->extra_tx_headroom = 4;
 
 	hw->wiphy->available_antennas_rx = BIT(ah->caps.max_rxchains) - 1;
 	hw->wiphy->available_antennas_tx = BIT(ah->caps.max_txchains) - 1;
