--- a/service/validate.c
+++ b/service/validate.c
@@ -54,10 +54,10 @@
 			continue;
 
-		o = json_object_object_get(r, v->package);
+		json_object_object_get_ex(r, v->package, &o);
 		if (!o) {
 			o = json_object_new_object();
 			json_object_object_add(r, v->package, o);
 		}
-		t = json_object_object_get(o, v->type);
+		json_object_object_get_ex(o, v->type, &t);
 		if (!t) {
 			t = json_object_new_object();
